### PR TITLE
Feature/#63 rule administration [WIP]

### PR DESCRIPTION
**Pull request by Neojume**
Request of AlertRule details and deletion of AlertRules.

### CODE REVIEW

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 1]
```diff
@@ -20,51 +21,82 @@ class LogParserActor extends Actor with ActorLogging {
...
+    }
   }
 
-  private def handleSubmitLogs(msg: SubmitLogs, metrics: List[Metric], metricToAlertActions: Map[MetricKey, List[ActorRef]]) {
+  private def handleDeleteRules(metrics: List[Metric], alertsByMetric: Map[MetricKey, List[ActorRef]], msg: DeleteAllAlertingRules) = {
+    log.debug("received delete request for all rules on metric {}", msg.metricKey)
+
+    alertsByMetric.get(msg.metricKey) match {
+      case None =>
+        sender() ! AlertRulesDeleted(msg.componentId)
```
Comment (mihaelaoprea) [inline src/main/scala/performanceanalysis/LogParserActor.scala]: might be handy to add metricKey as well to the AlertRulesDeleted msg

File: src/main/scala/performanceanalysis/administrator/Administrator.scala [hunk 2]
```diff
@@ -85,6 +87,25 @@ class Administrator(logReceiverActor: ActorRef) extends Server {
...
+        Future(HttpResponse(status = StatusCodes.NoContent))
+      case msg: MetricNotFound =>
+        Future(HttpResponse(status = NotFound))
+    }
+  }
+
+  private def handleGetAlertRules(resultFuture: Future[Any]): Future[HttpResponse] = {
+    resultFuture.flatMap {
+      case AlertRulesDetails(details) =>
+        Future(HttpResponse())
```
Comment (mihaelaoprea) [inline src/main/scala/performanceanalysis/administrator/Administrator.scala]: could return the details in the response. I guess this is on your todo list

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 3]
```diff
@@ -20,51 +21,82 @@ class LogParserActor extends Actor with ActorLogging {
...
+        }
+
+        context.become(normal(metrics, alertsByMetric - msg.metricKey))
+        sender() ! AlertRulesDeleted(msg.componentId)
+    }
+  }
+
+  private def handleSubmitLogs(msg: SubmitLogs, metrics: List[Metric], alertsByMetric: Map[MetricKey, List[ActorRef]]) {
     log.debug("received {} in {}", msg, self.path)
     for (metric <- metrics) {
```
Comment (Seetaramayya) [inline src/main/scala/performanceanalysis/LogParserActor.scala]: Is it really required `for` in `for`, is not possible to do with single `for` expression

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 4]
```diff
@@ -20,51 +21,82 @@ class LogParserActor extends Actor with ActorLogging {
...
-  private def handleSubmitLogs(msg: SubmitLogs, metrics: List[Metric], metricToAlertActions: Map[MetricKey, List[ActorRef]]) {
+  private def handleDeleteRules(metrics: List[Metric], alertsByMetric: Map[MetricKey, List[ActorRef]], msg: DeleteAllAlertingRules) = {
+    log.debug("received delete request for all rules on metric {}", msg.metricKey)
+
+    alertsByMetric.get(msg.metricKey) match {
+      case None =>
+        sender() ! AlertRulesDeleted(msg.componentId)
+      case Some(ruleList) =>
+        for (ruleActor <- ruleList) {
+          context.stop(ruleActor)
```
Comment (Seetaramayya) [inline src/main/scala/performanceanalysis/LogParserActor.scala]: Which one is better `stop` or `PoisnPill`

**Pull request merged**